src/NetscriptFunctions.ts

```diff
@@ -298,24 +298,23 @@ export const ns: InternalAPI<NSFull> = {
       }
       const moneyBefore = server.moneyAvailable;
       const growth = processSingleServerGrowth(server, threads, host.cpuCores);
-      const growthPercentForLogging = growth !== 0 ? growth - 1 : 0;
       const moneyAfter = server.moneyAvailable;
       ctx.workerScript.scriptRef.recordGrow(server.hostname, threads);
       const expGain = calculateHackingExpGain(server, Player) * threads;
       helpers.log(
         ctx,
         () =>
-          `Available money on '${server.hostname}' grown by ${formatPercent(
-            growthPercentForLogging,
-            6,
-          )}. Gained ${formatExp(expGain)} hacking exp (t=${formatThreads(threads)}).`,
+          `Available money on '${server.hostname}' grown by ${formatPercent(growth - 1, 6)}. Gained ${formatExp(
+            expGain,
+          )} hacking exp (t=${formatThreads(threads)}).`,
       );
       ctx.workerScript.scriptRef.onlineExpGained += expGain;
       Player.gainHackingExp(expGain);
       if (stock) {
         influenceStockThroughServerGrow(server, moneyAfter - moneyBefore);
       }
-      return Promise.resolve(growth);
+      // growth can only be 1 if server.moneyMax is 0. In that case, this API returns 0.
+      return Promise.resolve(growth !== 1 ? growth : 0);
```

Comment:
Ah hell. I'm not certain that this is true. For instance, if you try to grow when already at max money, I'm 95% sure you'll get to here and get 1 as the value of grow.

The real question is, is it fine to return 1 (instead of 0) in this case? It's a weird case because we throw for other "invalid" servers, but return a value here. But probably someone is relying on it, sigh.
```suggestion
      // growth can only be 1 if server.moneyMax is 0. In that case, this API returns 0.
      return Promise.resolve(growth !== 1 || server.moneyMax !== 0 ? growth : 0);
```

Reply:
You are right. I forgot the case in which the player grows a server that has already been at max money.

For "return 1 vs 0", I think returning 1 is also fine. It's sad that this behavior is another case of "legacy behavior".